doc/samples/SqlConfigurableRetryLogic_StepByStep_CustomProvider.cs

```diff
@@ -0,0 +1,254 @@
+        private readonly TimeSpan _maxValue = TimeSpan.FromMinutes(4);
+
+        public CustomEnumerator(TimeSpan timeInterval, TimeSpan maxTime, TimeSpan minTime)
+            : base(timeInterval, maxTime, minTime) {}
+
+        // Return fixed time on each request
+        protected override TimeSpan GetNextInterval()
+        {
+            return GapTimeInterval;
+        }
+
+        // Override the validate method to apply the new time range validation
+        protected override void Validate(TimeSpan timeInterval, TimeSpan maxTimeInterval, TimeSpan minTimeInterval)
+        {
+            if (minTimeInterval < TimeSpan.Zero || minTimeInterval > _maxValue)
+            {
+                throw new ArgumentOutOfRangeException(nameof(minTimeInterval));
+            }
+
+            if (maxTimeInterval < TimeSpan.Zero || maxTimeInterval > _maxValue)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maxTimeInterval));
+            }
+
+            if (timeInterval < TimeSpan.Zero || timeInterval > _maxValue)
+            {
+                throw new ArgumentOutOfRangeException(nameof(timeInterval));
+            }
+
+            if (maxTimeInterval < minTimeInterval)
+            {
+                throw new ArgumentOutOfRangeException(nameof(minTimeInterval));
+            }
+        }
+    }
+    // </Snippet6>
+
+    // <Snippet7>
+    public class CustomRetryLogic : SqlRetryLogicBase
+    {
+        // Maximum number of acceptable attempts
+        private const int maxAttempts = 20;
+
+        public CustomRetryLogic(int numberOfTries,
+                                 SqlRetryIntervalBaseEnumerator enumerator,
+                                 Predicate<Exception> transientPredicate)
+        {
+            if (!(numberOfTries > 0 && numberOfTries <= maxAttempts))
+            {
+                // 'numberOfTries' should be between 1 and 20.
+                throw new ArgumentOutOfRangeException(nameof(numberOfTries));
+            }
+
+            // Assign the given parameters to the relevant properties
+            NumberOfTries = numberOfTries;
+            RetryIntervalEnumerator = enumerator;
+            TransientPredicate = transientPredicate;
+            Current = 0;
+        }
+
+        // Prepare this object for the next round
+        public override void Reset()
+        {
+            Current = 0;
+            RetryIntervalEnumerator.Reset();
+        }
+
+        public override bool TryNextInterval(out TimeSpan intervalTime)
+        {
+            intervalTime = TimeSpan.Zero;
+            // First try has occurred before starting the retry process. 
+            // Check if it's still allowed to do retry
+            bool result = Current < NumberOfTries - 1;
+
+            if (result)
+            {
+                // Increase the number of attempts
+                Current++;
+                // It doesn't matter if the enumerator gets to the last value until the number of attempts end.
```

Comment:
This statement is confusing. What is it trying to say?

I'm guessing:
??? "It's okay if the enumerator doesn't get to the last value by the time we've reach our maximum number of attempts." ???

Reply:
`MoveNext()` returns `false` when it reached the last sequence of interval-times. Here I've just assessed the maximum number of attempts, and it doesn't matter if it is repeating a delay-time, multiple times.
I hope my explanation made it a bit more clear.